autoscheduler/scheduler/utils.py

```diff
@@ -0,0 +1,70 @@
+from datetime import time
+from functools import reduce
+from operator import mul
+from random import sample
+from typing import Any, Iterable, List, NamedTuple, Tuple, Union
+
+def random_product(*iterables: Iterable[Iterable], limit=100_000) -> Tuple[Any]:
```

Comment:
Just a thought, but I thought it might be nice if we somehow used the `CourseFilter` input to be the seed for this, so if on the frontend they did the exact same scheduling-generation query twice it would return the same thing